edgelet/support-bundle/src/support_bundle old.rs

```diff
@@ -0,0 +1,684 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use std::env;
+use std::ffi::OsString;
+use std::fs::File;
+use std::io::{Cursor, Seek};
+use std::path::Path;
+use std::process::Command as ShellCommand;
+
+use chrono::{DateTime, Local, NaiveDateTime, Utc};
+use failure::Fail;
+use futures::{Future, Stream};
+use tokio::prelude::*;
+use zip::{write::FileOptions, CompressionMethod, ZipWriter};
+
+use edgelet_core::{LogOptions, LogTail, Module, ModuleRuntime};
+
+use crate::error::{Error, ErrorKind};
+use crate::logs::pull_logs;
+
+pub fn make_bundle<M>(
+    output_location: OutputLocation,
+    log_options: LogOptions,
+    include_ms_only: bool,
+    verbose: bool,
+    iothub_hostname: Option<String>,
+    runtime: M,
+) -> Box<dyn Future<Item = (Box<dyn Read + Send>, u64), Error = Error> + Send>
+where
+    M: 'static + ModuleRuntime + Clone + Send + Sync,
+{
+    match output_location {
+        OutputLocation::File(location) => {
+            let writer = future::result(
+                File::create(Path::new(&location))
+                    .map_err(|err| Error::from(err.context(ErrorKind::SupportBundle))),
+            );
+
+            let state = writer.and_then(move |writer| {
+                make_state(
+                    log_options,
+                    include_ms_only,
+                    verbose,
+                    iothub_hostname,
+                    runtime,
+                    writer,
+                )
+            });
+
+            let bundle = state.and_then(BundleState::bundle_all);
+
+            let read =
+                bundle.and_then(|mut bundle| -> Result<(Box<dyn Read + Send>, u64), Error> {
+                    let result: Box<dyn Read + Send> = Box::new(
+                        bundle
+                            .zip_writer
+                            .finish()
+                            .map_err(|err| Error::from(err.context(ErrorKind::SupportBundle)))?,
+                    );
+                    Ok((result, 0)) // TODO: Get Size
```

Comment:
I don't know if we do this, but might be worthwhile to put a note about this somewhere in tracked items lest we forget about this :P

Reply:
also old file